webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/DataGridContextMenu/DataGridContextMenuFilter/DataGridContextMenuFilterService.ts

```diff
@@ -80,7 +83,8 @@ export class DataGridContextMenuFilterService {
     isHidden?: (context: IMenuContext<IDataGridCellMenuContext>) => boolean,
   ): Array<IContextMenuItem<IDataGridCellMenuContext>> {
     const { model, resultIndex, key } = context.data;
-    const data = model.source.getAction(resultIndex, ResultSetDataAction);
+    const source = model.source as unknown as ResultSetDataSource;
```

Comment:
<img width="1090" alt="Screenshot 2024-07-15 at 9 42 19 PM" src="https://github.com/user-attachments/assets/3b3291ca-eeb8-4202-aede-eb212f112f59">
in part of these places i don't repeat checks because it's exists already